File: 6JavaConsole/src/program/OldestFolderSearcher.java
package program;
import java.io.File;

public class OldestFolderSearcher 
{
	private String oldestDirName;
	private long oldestDirTime;
	
	public OldestFolderSearcher()
	{
		oldestDirName = "";
		oldestDirTime = Long.MAX_VALUE;
	}
	
	private void Search(File folder)
	{
		for (File fileEntry : folder.listFiles())
		{
	        if (fileEntry.isDirectory()) 
	        {
	        	if(fileEntry.lastModified() < oldestDirTime)
	        	{
	        		oldestDirName = fileEntry.getAbsolutePath();
	        		oldestDirTime = fileEntry.lastModified();
	        	}
	        	Search(fileEntry);
	        }
	    }
	}
	
	public String FindOldestDir(String path)
	{
		File folder = new File(path);
		Search(folder);
		return oldestDirName;
	}
}

File: DotNetCore/DotNetCore/OldestFolderSearcher.cs
using System;
using System.IO;

namespace DotNetCore
{
    public class OldestFolderSearcher
    {
        private string oldestDirName;
        private DateTime oldestDirDate;

        public OldestFolderSearcher()
        {
            oldestDirName = string.Empty;
            oldestDirDate = DateTime.MaxValue;
        }

        private void Search(string path)
        {
            DirectoryInfo info = new DirectoryInfo(path);
            foreach (var subDir in info.GetDirectories())
            {
                if (subDir.CreationTime < oldestDirDate)
                {
                    oldestDirDate = subDir.CreationTime;
                    oldestDirName = subDir.FullName;
                }
                Search(subDir.FullName);
            } 
        }

        public string FindOldestDir(string path)
        {
            Search(path);
            return oldestDirName;
        }
    }
}